src/components/nav-icon-item/nav-icon-item.tsx

```diff
@@ -0,0 +1,63 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, FC, MouseEventHandler, ReactNode } from 'react'
+import { ElAnchorNavIconItem, ElButtonNavIconItem } from './styles'
+
+interface CommonNavIconItemProps {
+  /**
+   * The JSX element to render as the icon
+   */
+  icon: ReactNode
+
+  /**
+   * Whether the nav item is active
+   * @default false
+   **/
+  isActive?: boolean
+}
+
+interface NavIconItemAsButtonElementProps extends CommonNavIconItemProps, ButtonHTMLAttributes<HTMLButtonElement> {
+  href?: never
+  target?: never
+  ref?: never
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+
+interface NavIconItemAsAnchorElementProps extends CommonNavIconItemProps, AnchorHTMLAttributes<HTMLAnchorElement> {
+  href?: string
+  target?: string
+  rel?: string
+  onClick?: MouseEventHandler<HTMLAnchorElement>
+}
+
+export type NavIconItemProps = NavIconItemAsButtonElementProps | NavIconItemAsAnchorElementProps
+
+const isButtonElement = (props: NavIconItemProps): props is NavIconItemAsButtonElementProps => {
+  return !props.href
+}
+
+export const NavIconItem: FC<NavIconItemProps> = (props) => {
+  if (!isButtonElement(props)) {
+    const { isActive, onClick, icon, ...rest } = props ?? {}
+
+    return (
+      <ElAnchorNavIconItem
+        {...rest}
+        role="button"
+        data-state={isActive ? 'active' : undefined}
```

Comment:
**suggestion:** I think [aria-current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) may be preferable; specifically when `isActive` is true, an anchor element would have `aria-current="page"` or a button element would have `aria-current="true"`.

Would also be reasonable to have a separate data attribute if it somehow makes the CSS styling simpler to achieve, but I suspect we could just style based on the presence of `aria-current` 🤔

Reply:
thanks for the suggestion, I think using `aria-current` is also a good option, as the styling is already separated in a styled component

with `aria-current`, it would simplifying the required custom attribute and improving accessibility at once